serialization.go

```diff
@@ -79,20 +79,36 @@ type Sender func(http.ResponseWriter, *http.Request, any) error
 
 // Send sends a response.
 // The format is determined by the Accept header.
-func Send(w http.ResponseWriter, r *http.Request, ans any) error {
-	switch parseAcceptHeader(r.Header.Get("Accept"), ans) {
-	case "application/xml":
-		return SendXML(w, nil, ans)
-	case "text/html":
-		return SendHTML(w, r, ans)
-	case "text/plain":
-		return SendText(w, nil, ans)
-	case "application/json":
-		return SendJSON(w, nil, ans)
-	case "application/x-yaml", "text/yaml; charset=utf-8", "application/yaml": // https://www.rfc-editor.org/rfc/rfc9512.html
-		return SendYAML(w, nil, ans)
+// If Accept header `*/*` is found Send will Attempt to send
+// HTML, and then JSON.
+func Send(w http.ResponseWriter, r *http.Request, ans any) (err error) {
+	for _, header := range parseAcceptHeader(r.Header) {
+		switch inferAcceptHeader(header, ans) {
+		case "application/xml":
+			err = SendXML(w, nil, ans)
+		case "text/html":
+			err = SendHTML(w, r, ans)
+		case "text/plain":
+			err = SendText(w, nil, ans)
+		case "application/json":
+			err = SendJSON(w, nil, ans)
+		case "application/x-yaml", "text/yaml; charset=utf-8", "application/yaml": // https://www.rfc-editor.org/rfc/rfc9512.html
+			err = SendYAML(w, nil, ans)
+		}
+
+		if err != nil {
+			continue
+			// should we log and continue?
+			// or
+			// if err == nil we just return?
+		}
+		return nil
+	}
+
+	if err != nil {
+		return err
 	}
-	return errors.New("unsupported Accept header")
+	return errors.New("no supported Accept header was not provided from: " + r.Header.Get("Accept"))
```

Comment:
I'm sorry but I'm doubtful about your algorithm

First, if you have a header you don't know

Then outside the loop you are testing if you have an error something that couldn't happen

I would like to suggest you this

```go
func Send(w http.ResponseWriter, r *http.Request, ans any) (err error) { 
         for _, header := range parseAcceptHeader(r.Header) { 
                 switch inferAcceptHeader(header, ans) { 
                 case "application/xml": 
                         return SendXML(w, nil, ans) 
                 case "text/html": 
                         return SendHTML(w, r, ans) 
                 case "text/plain": 
                        return SendText(w, nil, ans) 
                 case "application/json": 
                         return SendJSON(w, nil, ans) 
                 case "application/x-yaml", "text/yaml; charset=utf-8", "application/yaml": // https://www.rfc-editor.org/rfc/rfc9512.html 
                         return SendYAML(w, nil, ans) 
                 } 
         } 
 
         return errors.New("no supported Accept header was not provided from: " + r.Header.Get("Accept")) 
  }
```

There is no need to continue:
- if it works
- if there is an error, something that break the JSON is likely to break the HTML. A database error is likely to happen for each content type
- also, if the JSON is broken, you should stop and not try another content type, you would have silent errors that will fill your log, and people asking for a json,yaml,html via header, might be disappointed to receive a yaml if json fails. The error handling will be a nightmare on their side.

For this reason, using return should be preferred.

If you want to log you can log in the caller

Reply:
My comment sounds like what @EwenQuim reported [here](https://github.com/go-fuego/fuego/pull/173#discussion_r1734957641)

I'm only see what was posted in other PR now

Reply:
I think you're misunderstanding the original point. 

The case this is trying solve is when a caller passes something like `text/html, */*`, and the route they are returns only JSON. In that case with the algorithm you're proposing. We would just fail to respond with `text/html` and never attempt any other Accept types.

Reply:
Your second point 

> if there is an error, something that break the JSON is likely to break the HTML. A database error is likely to happen for each content type

Is handled by a controller returning and error. This code block wouldn't execute see: https://github.com/go-fuego/fuego/blob/main/serve.go#L113

Third point 

> also, if the JSON is broken, you should stop and not try another content type, you would have silent errors that will fill your log, and people asking for a json,yaml,html via header, might be disappointed to receive a yaml if json fails. The error handling will be a nightmare on their side.

if the caller is passing something like: `application/json, application/yaml` then we should try to fulfill whatever we can for the caller. Even if the caller asks for just `application/xml` on a route intending to return json and it cannot be fulfilled the HTTP spec [states](https://datatracker.ietf.org/doc/html/rfc2616#section-10.4.7):

> Note: HTTP/1.1 servers are allowed to return responses which are
      not acceptable according to the accept headers sent in the
      request. In some cases, this may even be preferable to sending a
      406 response. User agents are encouraged to inspect the headers of
      an incoming response to determine if it is acceptable.
      
This code only attempts to fulfill what the caller is asking for. If they are only asking for `application/json` then we will only attempt json. If the caller is asking for `application/json, */*`, the caller is giving permission to try anything that can possibly be returned; this is typically a browser.

Reply:
> I think you're misunderstanding the original point.

I notice now. I only read the previous PR after posting my initial comment here.

Reply:
I will say I do agree with you. If I as I caller pass `application/xml, */*` and I get a response of json on error. That would be strange, however the caller is telling us we're allowed to do that with that Accept Header.

Reply:
Too provide a bit more context from a browser if you're hitting a JSON endpoint you will get this:

![image](https://github.com/user-attachments/assets/e011cdce-2560-4de5-a20b-4f03514bbba5)

As we're failing to render any html.

Reply:
There is something strange anyway with the logic of handling of */*

I would say the order of the list is important.

If the user sends "application/rss, application/json", I would expect a json to be rendered

If the user sends "application/rss, */*", I would expect one supported format to be tried and an error to be returned if it fails

But then, it means that you need to define a priority between each content type handler.

I would expect */* to always try the exact same order (for example)

- application/json
- application/html

And that's it, not necessarily ALL possible handler. I mean no one should expect a YAML when someone asks */*

It's a "show me what you got" (Rick and Morty reference), the client expects to get interesting things.

Reply:
so `*/*` doesn't try all possible handlers. `*/*` is inferred based on the response type of the route. 

```golang
func InferAcceptHeaderFromType(ans any) string {
	_, ok := any(ans).(string)
	if ok {
		return "text/plain"
	}
	_, ok = any(ans).(*string)
	if ok {
		return "text/plain"
	}
	_, ok = any(ans).(HTML)
	if ok {
		return "text/html"
	}
	_, ok = any(ans).(CtxRenderer)
	if ok {
		return "text/html"
	}
	_, ok = any(&ans).(*CtxRenderer)
	if ok {
		return "text/html"
	}
	_, ok = any(ans).(Renderer)
	if ok {
		return "text/html"
	}
	_, ok = any(&ans).(*Renderer)
	if ok {
		return "text/html"
	}
	return "application/json"
}

func inferAcceptHeader(accept string, ans any) string {
	if accept == "" || accept == "*/*" {
		return InferAcceptHeaderFromType(ans)
	}
	return accept
}
```

Reply:
OK. It's clearer. I should have looked more carefully at the code